code/modules/projectiles/guns/projectile/pistol/lamia.dm

```diff
@@ -1,123 +1,122 @@
+	resultvars = list(/obj/item/gun/projectile/lamia)
+	gripvars = list(/obj/item/part/gun/grip/rubber)
+	mechanismvar = /obj/item/part/gun/mechanism/pistol
+	barrelvars = list(/obj/item/part/gun/barrel/kurtz)
+
+/obj/item/gun/projectile/lamia/update_icon()
+	..()
+
+	if(!ammo_magazine)
+		icon_state = initial(icon_state)
+	else if(!ammo_magazine.stored_ammo.len)
+		icon_state = initial(icon_state) + "_empty"
+	else
+		icon_state = initial(icon_state) + "_full"
+
+/obj/item/gun/projectile/lamia/gemini
+	name = "\"Gemini\" advanced heavy pistol"
+	desc = "An upgraded Lamia pistol fitted with a smart scope that harmonizes with neural link software. Chambered in 12mm, with an underbarrel LAM."
+	icon_state = "scoped_dark_lamia"
+	item_state = "scoped_dark_lamia"
+	matter = list(MATERIAL_PLASTEEL = 20, MATERIAL_PLASTIC = 10, MATERIAL_GLASS = 5, MATERIAL_SILVER = 3)
+	zoom_factors = list(0.4)
+	init_recoil = HANDGUN_RECOIL(0.7)
+	damage_multiplier = 1.3
+	price_tag = 1200
+	auto_eject = 1
+	auto_eject_sound = 'sound/weapons/smg_empty_alarm.ogg'
+	blacklist_upgrades = list(/obj/item/tool_upgrade/refinement/laserguide = TRUE) // Already has an underbarrel one, hence the recoil control
+	serial_type = "Sol Fed"
+	serial_shown = TRUE
+
+/obj/item/gun/projectile/lamia/scoped
+	name = "\"Argos\" advanced heavy pistol"
+	desc = "An upgraded Lamia pistol fitted with a smart scope that harmonizes with neural link software. Chambered in 12mm, with an underbarrel LAM - and made for private users."
+	icon_state = "scoped_lamia"
+	item_state = "scoped_lamia"
+	matter = list(MATERIAL_PLASTEEL = 20, MATERIAL_PLASTIC = 10, MATERIAL_GLASS = 5, MATERIAL_SILVER = 5)
+	auto_eject = 1
+	auto_eject_sound = 'sound/weapons/smg_empty_alarm.ogg'
+	zoom_factors = list(0.4)
+	init_recoil = HANDGUN_RECOIL(0.7)
+	price_tag = 1100
+	unload_sound 	= 'sound/weapons/guns/interact/avasarala_unload.ogg'
+	reload_sound 	= 'sound/weapons/guns/interact/hpistol_magin.ogg'
+	cocked_sound 	= 'sound/weapons/guns/interact/avasarala_cock.ogg'
+	fire_sound 	= 'sound/weapons/guns/fire/avasarala_fire.ogg'
+	serial_type = "SD GmbH"
+
+/obj/item/gun/projectile/lamia/socom
+	name = "\"SOCOM\" infiltration pistol"
+	desc = "An integrally suppressed Lamia pistol made for infiltrators and shock troopers. Chambered in 12mm, with an underbarrel LAM."
+	icon_state = "socom"
+	item_state = "socom"
+	matter = list(MATERIAL_PLASTEEL = 25, MATERIAL_PLASTIC = 10, MATERIAL_GLASS = 5, MATERIAL_SILVER = 3)
+	init_recoil = HANDGUN_RECOIL(0.1)
+	zoom_factors = list(0.5)
+	silenced = TRUE
+	price_tag = 1400
+	fire_sound_silenced = 'sound/weapons/guns/fire/vintorez.ogg'
+	serial_type = "Sol Fed"
+	blacklist_upgrades = list(/obj/item/tool_upgrade/refinement/laserguide = TRUE) // Already has an underbarrel one, hence the recoil control
+
+/obj/item/gun/projectile/lamia/amnesty
+	name = "\"Amnesty\" heavy pistol"
+	desc = "An in-colony variant of the Lamia pistol. Chambered in 12mm."

```

Comment:
Might want to add a little bit of the description from the Lamia here, otherwise people aren't going to encounter it much.